blockly/src/utils/pathfinding/Node.java

```diff
@@ -0,0 +1,61 @@
+package utils.pathfinding;
+
+import core.level.utils.Coordinate;
+import java.util.List;
+import utils.LevelUtils;
+
+/**
+ * A Node represents a point in the pathfinding grid.
+ *
+ * <p>Each node has a coordinate and a list of its walkable neighbors. This class is used in
+ * pathfinding algorithms to represent the state of the search.
+ *
+ * <p>It provides methods to access the coordinate and neighbors of the node.
+ *
+ * @see PathfindingLogic
+ */
+public class Node {
+  private final Coordinate coordinate;
+  private final List<Coordinate> neighbors;
+
+  /**
+   * Creates a new Node with the specified coordinate.
+   *
+   * @param coordinate The coordinate of the node.
+   */
+  public Node(Coordinate coordinate) {
+    this.coordinate = coordinate;
+    neighbors = LevelUtils.walkableNeighbors(coordinate);
```

Comment:
Du speicherst jetzt für jeden einzelnen Node die Liste der betretbaren Nachbarn. D.h. einige Nachbarn werden mehrfach gespeichert (über die Nodes hinweg gesehen).

Üblicherweise wird das erst beim Betrachten eines Nodes in der Hauptschleife der Suche berechnet und nicht gespeichert: 

```
So lange Queue nicht leer:
    Nimm ersten Knoten aus der Datenstruktur
    Falls Zielknoten: return "Gefunden"
    Else: 
        Markiere aktuellen Knoten als bearbeitet
        Schleife über alle betretbaren Nachbarn von aktuellem Knoten:
            Füge Nachbarn in Queue ein (außer er ist bereits markiert)
return "Failure"
```